src/main/java/seedu/pill/util/ItemMap.java

```diff
@@ -0,0 +1,335 @@
+
+        assert name != null && !name.trim().isEmpty() : "Item name cannot be null or empty";
+        assert quantity > 0 : "Quantity must be positive";
+
+        if (name == null || name.trim().isEmpty() || quantity <= 0) {
+            LOGGER.warning("Attempt to silently add invalid item: name=" + name + ", quantity=" + quantity);
+            return;
+        }
+
+        if (items.containsKey(name)) {
+            TreeSet<Item> itemSet = items.get(name);
+            boolean itemUpdated = false;
+            for (Item item : itemSet) {
+                if (item.getExpiryDate().equals(expiryDate)) {
+                    int newQuantity = item.getQuantity() + quantity;
+                    item.setQuantity(newQuantity);
+                    itemUpdated = true;
+                    LOGGER.fine("Silently updated existing item: " + name + ", new quantity: " + newQuantity);
+                    break;
+                }
+            }
+            if (!itemUpdated) {
+                itemSet.add(newItem);
+                LOGGER.fine("Silently added new item: " + newItem);
+            }
+        } else {
+            TreeSet<Item> itemSet = new TreeSet<>();
+            itemSet.add(newItem);
+            items.put(name, itemSet);
+            LOGGER.fine("Silently added new item: " + newItem);
+        }
+    }
+
+    /**
+     * Deletes an item from the list by its name.
+     *
+     * @param name       The name of the item to be deleted.
+     * @param expiryDate The date of the item to be deleted.
+     */
+    public void deleteItem(String name, Optional<LocalDate> expiryDate) {
+        assert name != null : "Item name cannot be null";
+
+        if (name == null || name.trim().isEmpty()) {
+            LOGGER.warning("Attempt to delete item with invalid name: " + name);
+            System.out.println("Invalid item name.");
+            return;
+        }
+
+        TreeSet<Item> itemSet = items.get(name);
+        if (itemSet != null) {
+            Item dummyItem = expiryDate.map(ex -> new Item(name, 0, ex))
+                            .orElse(new Item(name, 0));
+            Item removedItem = itemSet.ceiling(dummyItem);
+            if (removedItem != null && removedItem.getExpiryDate().equals(expiryDate)) {
+                itemSet.remove(removedItem);
+                LOGGER.info("Deleted item: " + removedItem);
+                System.out.println("Deleted the following item from the inventory: \n"
+                        + removedItem);
+                if (itemSet.isEmpty()) {
+                    items.remove(name);
+                }
+            } else {
+                LOGGER.warning("Attempt to delete non-existent item: " + removedItem);
+                System.out.println("Item not found: " + removedItem);
+            }
+        } else {
+            LOGGER.warning("Attempt to delete non-existent item: " + name);
+            System.out.println("Item not found: " + name);
+        }
```

Comment:
Maybe you can make the [happy path more obvious](https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) by switching this into guard clauses instead?